examples/terraform/aws-dualstack/main.tf

```diff
@@ -0,0 +1,414 @@
+  })
+}
+
+resource "aws_security_group_rule" "ingress_self_allow_all" {
+  type              = "ingress"
+  security_group_id = aws_security_group.common.id
+
+  description = "allow all incoming traffic from members of this group"
+  from_port   = 0
+  to_port     = 0
+  protocol    = "-1"
+  self        = true
+}
+
+resource "aws_security_group_rule" "egress_allow_all" {
+  type              = "egress"
+  security_group_id = aws_security_group.common.id
+
+  description = "allow all outgoing traffic"
+  from_port   = 0
+  to_port     = 0
+  protocol    = "-1"
+  cidr_blocks = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "nodeports" {
+  type              = "ingress"
+  security_group_id = aws_security_group.common.id
+
+  description = "open nodeports"
+  from_port   = 30000
+  to_port     = 32767
+  protocol    = "tcp"
+  cidr_blocks = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group" "elb" {
+  name        = "${var.cluster_name}-api-lb"
+  description = "kube-api firewall"
+  vpc_id      = data.aws_vpc.selected.id
+
+  egress {
+    description = "allow all outgoing traffic"
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    description = "allow anyone to connect to tcp/6443"
+    from_port   = 6443
+    to_port     = 6443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    description = "allow all outgoing traffic"
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  ingress {
+    description = "allow anyone to connect to tcp/6443"
+    from_port   = 6443
+    to_port     = 6443
+    protocol    = "tcp"
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = tomap({
+    "Cluster" = var.cluster_name,
+  })
+}
+
+resource "aws_security_group" "ssh" {
+  name        = "${var.cluster_name}-ssh"
+  description = "ssh access"
+  vpc_id      = data.aws_vpc.selected.id
+
+  ingress {
+    description = "allow incoming SSH"
+    from_port   = var.ssh_port
+    to_port     = var.ssh_port
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    description = "allow incoming SSH IPv6"
+    from_port   = var.ssh_port
+    to_port     = var.ssh_port
+    protocol    = "tcp"
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = tomap({
+    "Cluster" = var.cluster_name,
+  })
+}
+
+resource "aws_security_group_rule" "egress_allow_all_ipv6"  {
+  type              = "egress"
+  security_group_id = aws_security_group.common.id
+
+  description = "allow all outgoing traffic"
+  from_port   = 0
+  to_port     = 0
+  protocol    = "-1"
+  ipv6_cidr_blocks = ["::/0"]
+}
+
+resource "aws_security_group_rule" "nodeports_ipv6"  {
+  type              = "ingress"
+  security_group_id = aws_security_group.common.id
+
+  description = "open nodeports"
+  from_port   = 30000
+  to_port     = 32767
+  protocol    = "tcp"
+  ipv6_cidr_blocks = ["::/0"]
+}
```

Comment:
Nit: let's colocate relevant rules. For example, let's put this rule below nodeport rule for IPv4.